src/main/kotlin/io/snyk/plugin/SnykPostStartupActivity.kt

```diff
@@ -95,6 +105,10 @@ class SnykPostStartupActivity : ProjectActivity {
         if (isContainerEnabled()) {
             getKubernetesImageCache(project)?.scanProjectForKubernetesFiles()
         }
+
+        ExtensionPointsUtil.controllerLifecycleManager.extensionList.forEach {
+            it.register(SnykControllerImpl(project))
```

Comment:
This is quite neat. Should we maybe use a singleton implementation?

Reply:
I think we might benefit from a singleton here if we found the controller needed to become stateful and we wanted to share that state among all the extensions. Or if we were hyper-optimizing for every byte of heap?

Conversely, choosing a singleton now might conflict with a future need for per-extension state and we'd have to change it.

I'm +/- 0 on this one... we can always change the implementation later if we find we need to?

Reply:
Yeah, fine with me.